app/main/routes_picoferm_api.py

```diff
@@ -32,10 +34,31 @@ def process_ferm_registered(args):
 @main.route('/API/PicoFerm/checkFirmware')
 @use_args(check_ferm_firmware_args, location='querystring')
 def process_check_ferm_firmware(args):
-    # TODO download the firmware for a picoferm and support minimum firmware versions
+    if firmware_upgrade_required(MachineType.PICOFERM, args['version']):
+        return '#1#'
     return '#0#'
 
 
+# Get Firmware: /API/pico/getFirmware?uid={UID}
+#     Response: RAW Bin File Contents
+get_firmware_args = {
+    'uid': fields.Str(required=True),       # 12 character alpha-numeric serial number
+}
+@main.route('/API/PicoFerm/getFirmwareAddress')
+@use_args(get_firmware_args, location='querystring')
+def process_get_firmware_address(args):
+    filename = firmware_filename(MachineType.PICOFERM, minimum_firmware(MachineType.PICOFERM))
+    return '#http://picobrew.com/firmware/picoferm/{}#'.format(filename)

```

Comment:
Should this be https to match the change done for the picostill?

Reply:
The Z is the only one that uses certs.. why?  If you didn't have a z and weren't running on the pi image we made, it wouldn't work

Reply:
Actually https doesn't even work from the ferm. Also as the comment states the still want intended to be https cause under certain setups that will also fail since https isn't always setup by the user.